vendor/github.com/docker/distribution/registry/client/blob_writer.go

```diff
@@ -99,7 +99,7 @@ func (hbu *httpBlobUpload) Write(p []byte) (n int, err error) {
 		return 0, fmt.Errorf("bad range format: %s", rng)
 	}
 
-	return (end - start + 1), nil
```

Comment:
Oh, it was because we use a MultiReader in order to calculate the digest and tarsum of the layer as we stream it.  MultiReader requires the nested stream to correctly return the write size, otherwise it returns ErrShortWrite